File: frontend/src/services/accountService.ts
import api from './api';
import type {
  ApiResponse,
  Account,
  AccountCreateRequest,
  AccountUpdateRequest,
  BrokerInfo,
  BrokerType,
} from '../types';

export const accountService = {
  getAccounts: async (brokerType?: BrokerType) => {
    const params = brokerType ? { brokerType } : {};
    const response = await api.get<ApiResponse<Account[]>>('/accounts', { params });
    return response.data;
  },

  getAccount: async (accountId: number) => {
    const response = await api.get<ApiResponse<Account>>(`/accounts/${accountId}`);
    return response.data;
  },

  createAccount: async (data: AccountCreateRequest) => {
    const response = await api.post<ApiResponse<Account>>('/accounts', data);
    return response.data;
  },

  updateAccount: async (accountId: number, data: AccountUpdateRequest) => {
    const response = await api.put<ApiResponse<Account>>(`/accounts/${accountId}`, data);
    return response.data;
  },

  deleteAccount: async (accountId: number) => {
    const response = await api.delete<ApiResponse<void>>(`/accounts/${accountId}`);
    return response.data;
  },

  getBrokers: async () => {
    const response = await api.get<ApiResponse<BrokerInfo[]>>('/brokers');
    return response.data;
  },
};

File: backend/src/main/java/com/stock/account/service/AccountService.java
package com.stock.account.service;

import com.stock.account.domain.BrokerType;
import com.stock.account.dto.AccountCreateRequest;
import com.stock.account.dto.AccountResponse;
import com.stock.account.dto.AccountUpdateRequest;

import java.util.List;

public interface AccountService {

    AccountResponse createAccount(Long userId, AccountCreateRequest request);

    List<AccountResponse> getAccounts(Long userId, BrokerType brokerType);

    AccountResponse getAccount(Long accountId, Long userId);

    AccountResponse updateAccount(Long accountId, Long userId, AccountUpdateRequest request);

    void deleteAccount(Long accountId, Long userId);
}
